maps.Rmd

```diff
@@ -82,49 +83,66 @@ The `coord_sf()` function governs the map projection, discussed in Section \@ref
+Adding labels to maps is an example of annotating plots (Chapter \@ref(annotations)) and is supported by `geom_sf_label()` and `geom_sf_text()`. For example, while an Australian audience might be reasonably expected to know the names of the Australian states (and are left unlabelled in the plot above) few Australians would know the names of different electorates in the Sydney metropolitan region. In order to draw an electoral map of Sydney, then, we would first need to extract the 
+map data for the relevant elextorates, and then add the label. The plot below zooms in on the Sydney region by specifying `xlim` and `ylim` in `coord_sf()` and then uses `geom_sf_label()` to overlay each electorate with a label:
 
 ```{r}
-sydney_map <- ozmaps::abs_ced %>%
-  filter(NAME %in% c("Sydney", "Wentworth", "Warringah", "Kingsford Smith", "Grayndler", "Lowe", "North Sydney", "Barton", "Bradfield", "Banks", "Blaxland", "Reid", "Watson", "Fowler", "Werriwa", "Prospect", "Parramatta", "Bennelong", "Mackellar", "Greenway", "Mitchell", "Chifley", "McMahon"))
-
+# filter electorates in the Sydney metropolitan region
+sydney_map <- ozmaps::abs_ced %>% filter(NAME %in% c(
+  "Sydney", "Wentworth", "Warringah", "Kingsford Smith", "Grayndler", "Lowe", 
+  "North Sydney", "Barton", "Bradfield", "Banks", "Blaxland", "Reid", 
+  "Watson", "Fowler", "Werriwa", "Prospect", "Parramatta", "Bennelong", 
+  "Mackellar", "Greenway", "Mitchell", "Chifley", "McMahon"
+))
+
+# draw the electoral map of Sydney
 ggplot(sydney_map) + 
   geom_sf(aes(fill = NAME), show.legend = FALSE) + 
   coord_sf(xlim = c(150.97, 151.3), ylim = c(-33.98, -33.79)) + 
   geom_sf_label(aes(label = NAME), label.padding = unit(1, "mm"))
 ```
 
+The warning message is worth noting. Internally `geom_sf_label()` uses the function
+`st_point_on_surface()` from the sf package to place labels, and the warning message 
+occurs because most algorithms used by sf to compute geometric quantities (e.g., centroids, 
+interior points) are based on an assumption that the points lie in on a flat two dimensional
+surface and parameterised with Cartesian co-ordinates. This assumption is not strictly 
+warranted, and in some cases (e.g., regions near the poles) calculations that treat
+longitude and latitude in this way will give erroneous answers. For this reason, the sf
+package produces warning messages when it relies on this approximation.
+
 <!-- HW: worth commenting on warning? "st_point_on_surface may not give correct results for longitude/latitude data" -->
+<!-- DN: is that the right answer? -->
```

Comment:
It's "right" as per sf-philosophy, but not to me - I just wanted a label, it falls in the polygon so it's perfectly fine. (How deep do you want to go, there's more than one kind of centroid and more than geographic perspectives) :)